content/test/unit/repository/DeploymentsRepository.spec.ts

```diff
@@ -0,0 +1,456 @@
+      db = mock(MockedDataBase)
+      repository = new DeploymentsRepository(instance(db) as any)
+
+      when(db.map(anything(), anything(), anything())).thenReturn(Promise.resolve(expectedResult))
+      result = await repository.getAmountOfDeployments()
+    })
+
+    it('should call the database with the expected query', () => {
+      verify(
+        db.map(`SELECT entity_type, COUNT(*) AS count FROM deployments GROUP BY entity_type`, deepEqual([]), anything())
+      ).once()
+    })
+
+    it('should return a Map with the result', () => {
+      expect(result).toEqual(new Map(expectedResult as any))
+    })
+  })
+
+  describe('getHistoricalDeployments', () => {
+    beforeEach(() => {
+      db = mock(MockedDataBase)
+      repository = new DeploymentsRepository(instance(db) as any)
+    })
+
+    describe("when it doesn't receive a lastId", () => {
+      beforeEach(() => {
+        db = mock(MockedDataBase)
+        repository = new DeploymentsRepository(instance(db) as any)
+      })
+
+      describe('when it receives a field or order to sort by', () => {
+        beforeEach(() => {
+          when(db.map(anything(), anything(), anything())).thenReturn(Promise.resolve([]))
+        })
+
+        it('should call the database with the expected sorting', async () => {
+          await repository.getHistoricalDeployments(
+            0,
+            10,
+            { from: 1, to: 11 },
+            {
+              field: SortingField.ENTITY_TIMESTAMP,
+              order: SortingOrder.ASCENDING
+            }
+          )
+
+          const args = capture(db.map).last()
+
+          expect(args[0]).toContain(`ORDER BY dep1.entity_timestamp ASC`)
+          expect(args[0]).toContain(
+            `dep1.entity_timestamp >= to_timestamp($(fromEntityTimestamp) / 1000.0) AND dep1.entity_timestamp <= to_timestamp($(toEntityTimestamp) / 1000.0)`
+          )
+          expect(args[1]).toEqual(jasmine.objectContaining({ fromEntityTimestamp: 1, toEntityTimestamp: 11 }))
+        })
+      })
+
+      describe("when it doesn't receive a field or order to sort by", () => {
+        beforeEach(() => {
+          when(db.map(anything(), anything(), anything())).thenReturn(Promise.resolve([]))
+        })
+
+        it('should call the database with the default sorting', async () => {
+          await repository.getHistoricalDeployments(0, 10, { from: 1, to: 11 })
+
+          const args = capture(db.map).last()
+
+          expect(args[0]).toContain(`ORDER BY dep1.local_timestamp DESC`)
+          expect(args[0]).toContain(
+            `dep1.local_timestamp >= to_timestamp($(fromLocalTimestamp) / 1000.0) AND dep1.local_timestamp <= to_timestamp($(toLocalTimestamp) / 1000.0)`
+          )
+          expect(args[1]).toEqual(jasmine.objectContaining({ fromLocalTimestamp: 1, toLocalTimestamp: 11 }))
+        })
+      })
+    })
+
+    describe('when it receives a lastId', () => {
+      const lastId = '1'
+
+      beforeEach(() => {
```

Comment:
As we don't really care about the response from the MockedDataBase can we configure this only once for the entire file?

Reply:
We can, but that makes the tests coupled since the mock is the same among all tests

Reply:
That's why I initialized it in each test context